OpenMarket/OpenMarket/Controller/RegisterViewController.swift

```diff
@@ -0,0 +1,170 @@
+        picker.delegate = self
+        setBaseImage()
+        navigationItem.title = Constant.navigationTitle
+    }
+}
+
+// MARK: - Method
+extension RegisterViewController {
+    func addImageToStackView(image: UIImage){
+        let imageView = UIImageView(frame: CGRect(origin: .zero, size: CGSize(width: 50, height: 50)))
+        imageView.isUserInteractionEnabled = true
+        imageView.image = image
+        let gesture = UISwipeGestureRecognizer(target: self, action: #selector(deleteImageView))
+        gesture.direction = .up
+        imageView.addGestureRecognizer(gesture)
+        
+        addImageHorizontalStackView.addLastBehind(view: imageView)
+        
+        NSLayoutConstraint.activate([
+            imageView.heightAnchor.constraint(equalTo: addImageHorizontalStackView.heightAnchor),
+            imageView.widthAnchor.constraint(equalTo: imageView.heightAnchor)
+        ])
+        
+        if addImageHorizontalStackView.arrangedSubviews.count == 6 {
+            addImageButton.isHidden = true
+        } else {
+            addImageButton.isHidden = false
+        }
+        addImageHorizontalStackView.setNeedsDisplay()
+    }
+    
+    func setBaseImage() {
+        addImageHorizontalStackView.addArrangedSubview(addImageButton)
+        
+        NSLayoutConstraint.activate([
+            addImageButton.heightAnchor.constraint(equalTo: addImageHorizontalStackView.heightAnchor),
+            addImageButton.widthAnchor.constraint(equalTo: addImageButton.heightAnchor)
+        ])
+    }
+    
+    func wrapperRegistrationParameter() -> RegistrationParameter? {
+        guard let name = nameTextField.text, name.count >= 3 else {
+            return nil
+        }
+        guard let descriptions = textView.text, descriptions.count < 1000 else {
+            return nil
+        }
+        guard let price = Double(priceTextField.text ?? "0"), price >= 0 else {
+            return nil
+        }
+        guard let selectedText = currencySegmentedControl.titleForSegment(at: currencySegmentedControl.selectedSegmentIndex), let currency = Currency(rawValue: selectedText) else {
+            return nil
+        }
+        guard let discountedPrice = Double(discountPriceTextField.text ?? "0"), discountedPrice >= 0 else {
+            return nil
+        }
+        guard let stock = Int(stockTextField.text ?? "0"), stock >= 0 else {
+            return nil
+        }
+        let secret = Secret.registerSecret
+        
+        return RegistrationParameter(name: name, descriptions: descriptions, price: price, currency: currency, discountedPrice: discountedPrice, stock: stock, secret: secret)
+    }
+    
+    func wrapperImage() -> [UIImage] {
+        var imageArray = [UIImage]()
+        
+        for subView in addImageHorizontalStackView.arrangedSubviews {
+            if let subView = subView as? UIImageView, let uiImage = subView.image {
+                imageArray.append(uiImage)
+            }
+        }
```

Comment:
배열을 다른 배열로 변환하는 로직은 `map`이나 `compactMap`을 사용해 더 간단하게 나타낼 수 있을 것 같아요.

Reply:
넵 고차함수를 사용해서 짧게 표현해보... 려 했는데 고차함수로 변경할시 filter와 map을 둘다 써야 되더군요! 그래서 바꾸지 않았습니다.

Reply:
어떤 방식으로 변경해보셨을까요? 👀 
코드를 보여드리는 건 정답을 제시하는 것으로 느끼실까봐 피하긴 하는데..
```suggestion
        let imageArray = addImageHorizontalStackView.arrangedSubviews.compactMap { ($0 as? UIImageView)?.image }
```
요렇게 바꿔볼 수 있을 것 같아요.

Reply:
수정완료하였습니다.